extensions/pkg/controller/dnsrecord/reconciler.go

```diff
@@ -235,3 +244,47 @@ func (r *reconciler) delete(ctx context.Context, dns *extensionsv1alpha1.DNSReco
 
 	return reconcile.Result{}, nil
 }
+
+func updateCreatedCondition(status extensionsv1alpha1.Status, conditionStatus gardencorev1beta1.ConditionStatus, reason string, updateIfExisting bool) error {
+	conditions := status.GetConditions()
+	index := len(conditions)
+	for i, c := range conditions {
+		if c.Type == createdConditionType {
+			if !updateIfExisting {
+				return nil
+			}
+			index = i
+			break
+		}
+	}
+	if index == len(conditions) {
+		conditions = append(conditions, gardencorev1beta1.Condition{Type: createdConditionType})
+	}
+	c := &conditions[index]
+	if c.Status == conditionStatus {
+		return nil
+	}
+	c.Status = conditionStatus
+	c.LastTransitionTime = metav1.Time{Time: time.Now()}
+	c.LastUpdateTime = metav1.Time{Time: time.Now()}
+	c.Reason = reason
+	status.SetConditions(conditions)
+	return nil
+}
+
+func getCreatedConditionStatus(status extensionsv1alpha1.Status) gardencorev1beta1.ConditionStatus {
+	for _, c := range status.GetConditions() {
```

Comment:
You could use `gardencorev1beta1helper.GetCondition` here.

Reply:
Don't save code here.